create.go

```diff
@@ -0,0 +1,14 @@
+package statusresouce
+
+import (
+	"context"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+
+	// TODO add version if not tracked
+	// TODO when version changes set updating status
+	// TODO when updating state is set and guest cluster is updated set updated status
```

Comment:
This is a bit more sophisticated as the implications here are also that we need an abstracted way to compute a k8s client for the guest cluster. This is also the reason why I decided not to put the status resource implementation here into `operatorkit` but go with its own repo, because it is quite opinionated towards Giant Swarm. Thus we also have to extract the k8s client magic we use in `cluster-operator` and provide two more ways of accessing the cluster ID and the guest API domain.

Reply:
Yes this looks like GS stuff. Makes sense to extract other repo. I thought me maybe want to have a common repo for various reusable resources. But we can move code later.

Reply:
Yes I like extracting and extending the guestcluster service in cluster-operator. Useful here and maybe also in the e2e tests.